diff --git a/distrib/build/Makefile b/distrib/build/Makefile
index 0e6d3c3..61b88f7 100644
--- a/distrib/build/Makefile
+++ b/distrib/build/Makefile
@@ -223,7 +223,7 @@ build: crumbs/host.checked crumbs/Final.built
 #
 crumbs/Debian.built:
 	@(tput smso; echo -e "\nDebian Build"; tput rmso) <$(BUILD_IN) >>$(BUILD_OUT) 2>&1
-	env -i toolcrib/make_debian
+	env -i toolcrib/make_debs
 	#@touch crumbs/debian.built
 
 debs: crumbs/Debian.built
@@ -390,7 +390,7 @@ clean:
 	  no_be=1 $(ARCHset) $(MAKE) --no-print-directory clean-sources; \
 	  no_be=1 $(MAKE) --no-print-directory clean-tools; \
 	  no_be=1 $(MAKE) --no-print-directory clean-crumbs; \
-#	  no_be=1 $(MAKE) --no-print-directory clean-debway; \
+	  no_be=1 $(MAKE) --no-print-directory clean-debway; \
 	  rm -f $(SRC_TGZ); \
 	  rm -f nohup.out; \
 	fi
@@ -398,21 +398,21 @@ clean:
 
 ##	Clan the Debian way
 clean-debway:
-#	cd sources/coretree; dpkg-buildpackage -Tclean
+	cd sources/coretree; dpkg-buildpackage -Tclean; echo; echo
 #	cd sources/libsmooth; dpkg-buildpackage -Tclean
-#	cd sources/libsmooth+setup; dpkg-buildpackage -Tclean
-#	cd sources/libsmooth+setup-git; make clean
-#	cd sources/smoothd; dpkg-buildpackage -Tclean
-#	cd sources/express; dpkg-buildpackage -Tclean
-	cd sources/httpd; dpkg-buildpackage -Tclean
-	cd sources/snort; dpkg-buildpackage -Tclean
-	cd sources/beep2; dpkg-buildpackage -Tclean
-	cd sources/setup; dpkg-buildpackage -Tclean
-	cd sources/setkerneltz; dpkg-buildpackage -Tclean
-	cd sources/grub; dpkg-buildpackage -Tclean
-	cd sources/trafficstats; dpkg-buildpackage -Tclean
-	cd sources/runtimeroot; dpkg-buildpackage -Tclean
-	cd sources/device-ids; dpkg-buildpackage -Tclean
+	cd sources/libsmooth+setup; dpkg-buildpackage -Tclean; echo; echo
+#	cd sources/libsmooth+setup-git; make clean; echo; echo
+#	cd sources/smoothd; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/express; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/httpd; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/snort; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/beep2; dpkg-buildpackage -Tclean; echo; echo
+#	cd sources/setup; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/setkerneltz; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/grub; dpkg-buildpackage -Tclean; echo; echo
+#	cd sources/trafficstats; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/runtimeroot; dpkg-buildpackage -Tclean; echo; echo
+	cd sources/device-ids; dpkg-buildpackage -Tclean; echo; echo
 
 
 # Empty the buildroot directory
diff --git a/distrib/build/sources/Makefile.conf b/distrib/build/sources/Makefile.conf
index b5e4047..230d1c4 100644
--- a/distrib/build/sources/Makefile.conf
+++ b/distrib/build/sources/Makefile.conf
@@ -7,16 +7,7 @@
 ifdef GLOBALDEBUG
 export GLOBALDEBUG = ON
 endif
-ifeq ($(BUILD_PHASE), Programs)
-SOURCES_DIR = $(DISTRIB_PATH)/sources
 
-TARGET_DIR = $(DISTRIB_PATH)/target
-
-DOWNLOADS_DIR = $(DISTRIB_PATH)/downloads
-
-DL_CMD = $(DISTRIB_PATH)/toolcrib/dlverify $(DOWNLOADS_DIR)
-
-else
 SOURCES_DIR = $(ROOT)/build/sources
 
 TARGET_DIR = $(ROOT)/build/target
@@ -24,7 +15,7 @@ TARGET_DIR = $(ROOT)/build/target
 DOWNLOADS_DIR = $(ROOT)/build/downloads
 
 DL_CMD = $(ROOT)/build/toolcrib/dlverify $(DOWNLOADS_DIR)
-endif
+
 EXTENSION = .tar.gz
 
 TARBALL = $(PACKAGE)-$(VERSION)
diff --git a/distrib/build/sources/Makefile.flags-arch b/distrib/build/sources/Makefile.flags-arch
index f9e9450..42569cd 100644
--- a/distrib/build/sources/Makefile.flags-arch
+++ b/distrib/build/sources/Makefile.flags-arch
@@ -68,7 +68,6 @@ export mARCH = $(shell uname -m | sed -e 's/i.86/$(ARCH32)/')
 endif
 
 export mTUPLE = $(mARCH)
-export LDFLAGS = -Wl,-O1 -Wl,--as-needed OPTIMIZE=-O2 -pipe
 
 # See what we can build; that is, see if the chosen mARCH is possible.
 # If building on a multilib host, both will be zero. If building on pure i686
diff --git a/distrib/build/sources/coretree/tree/httpd/html/ui/img/urlfilter/background.bak b/distrib/build/sources/coretree/tree/httpd/html/ui/img/urlfilter/background.bak
deleted file mode 100644
index 06298b6..0000000
Binary files a/distrib/build/sources/coretree/tree/httpd/html/ui/img/urlfilter/background.bak and /dev/null differ
diff --git a/distrib/build/toolcrib/Makefile.build_environment b/distrib/build/toolcrib/Makefile.build_environment
index 69f4652..0f08dae 100644
--- a/distrib/build/toolcrib/Makefile.build_environment
+++ b/distrib/build/toolcrib/Makefile.build_environment
@@ -17,6 +17,8 @@ prep_env:
 	  echo -e "export ROOT=\"\"\nexport KERNEL_TYPE=\"$(KERNEL_TYPE)\""; \
 	  echo -e "export BUILD_IN=\"$(BUILD_IN)\"\nexport BUILD_OUT=\"$(BUILD_OUT)\""; \
 	  echo -e "export ARCH32=\"$(ARCH32)\"\nexport ARCH64=\"$(ARCH64)\""; \
+	  echo -e "export DISTRIB_PATH=\"$(DISTRIB_PATH)\"\nexport LDFLAGS=\"$(LDFLAGS)\""; \
+	  echo -e "export ARCH=\"$(ARCH)\"\nexport PWD=\"$(PWD)\""; \
 	  echo -e "export TAR_OPTIONS=\"--no-same-owner --no-same-permissions\""; \
 	) | sort > crumbs/build_environment
 	@ # Also truncate the progress log (no-op if it's a tty)
diff --git a/distrib/build/toolcrib/environment b/distrib/build/toolcrib/environment
index cff25da..d75d881 100644
--- a/distrib/build/toolcrib/environment
+++ b/distrib/build/toolcrib/environment
@@ -151,6 +151,7 @@ function set_build_environment() {
    Debian)
       export mARCH = $(shell dpkg-architecture -qDEB_HOST_ARCH)
       export ROOT="/"
+    #  export ROOT="$DISTRIB_PATH"
       export PATH="$PATH"
       # placeholder to build PKG_ROOT
       export PKG_ROOT_PATH="$ROOT/build/buildroot"
@@ -158,10 +159,11 @@ function set_build_environment() {
       # PKG_DIR is the configure-like 'prefix' where the package ends up installed
       export PKG_DIR="/usr"
 
-      # Now report the environment
+      # Now report the environment for Debian
       #
-      echo "    Environment:"
-      echo "      Build Phase: $BUILD_PHASE; ARCH=$ARCH mARCH=$mARCH"
+      echo "    Debian Environment:"
+      echo "      Build Phase: $BUILD_PHASE"
+      echo "      ARCH=$ARCH mARCH=$mARCH mTUPLE=$mTUPLE"
       echo "      (PKG_ROOT not yet set); PKG_DIR=$PKG_DIR; "
       echo "      ROOT=$ROOT"
       echo "      PATH=$PATH; "
@@ -187,18 +189,19 @@ function set_build_environment() {
       export PKG_ROOT="/THIS-AINT-BEEN-SET-YET/"
       # PKG_DIR is the configure-like 'prefix' where the package ends up installed
       export PKG_DIR="/usr"
-
-
-      # PKG_DIR is the configure-like 'prefix' where the package ends up installed
-      export PKG_DIR="$TOOLS_DIR"
       ##	Gentoo seems to always set this, So We Gonna Too...
       export LDFLAGS="-Wl,-O1 -Wl,--as-needed OPTIMIZE=-O2 -pipe"
 
-      # Now report the environment
+      # Now report the environment for Programs
       #
-      echo "    Environment:"
-      echo "      Build Phase: $BUILD_PHASE; ARCH=$ARCH mARCH=$mARCH mTUPLE=$mTUPLE"
+      echo "    Programs Environment:"
+      echo "      Build Phase: $BUILD_PHASE"
+      echo "      ARCH=$ARCH mARCH=$mARCH mTUPLE=$mTUPLE"
       echo "      PKG_ROOT=$PKG_ROOT; PKG_DIR=$PKG_DIR; "
+      echo "      DISTRIB_PATH=$DISTRIB_PATH"
+      echo "      ROOT=$ROOT"
+      echo "      ROOT=$ROOT"
+      echo "      ROOT=$ROOT"
       echo "      ROOT=$ROOT"
       echo "      PATH=$PATH; "
       echo "      CFLAGS=$CFLAGS; "
